python/ray/serve/_private/deployment_state.py

```diff
@@ -2083,11 +2107,24 @@ def _check_startup_replicas(
                 self._deployment_scheduler.on_replica_running(
                     replica.replica_id, replica.actor_node_id
                 )
-                logger.info(
+
+                # Log the startup latency.
+                e2e_replica_start_latency = time.time() - replica._start_time
```

Comment:
Non-blocker: I'm not sure if there can ever be race condition between this getting called vs the replica is started? Seems like `_start_time` is not populated when the replica is initialized, but when `start()` is called. So there might be a possibility that `replica._start_time` is still None at this point? But then I guess this is within the `start_status == ReplicaStartupStatus.SUCCEEDED` block so maybe that's not possible. 🤔

Reply:
That's a good callout, but I don't think `replica._start_time` can be `None` at this point. It gets set in both `start()` and `recover()`, so by the time `start_status == ReplicaStartupStatus.SUCCEEDED`, it should be populated.